src/Umbraco.Web/Services/IconService.cs

```diff
@@ -91,5 +109,26 @@ private IconModel CreateIconModel(string iconName, string iconPath)
                 return null;
             }
         }
+
+        private IEnumerable<FileInfo> GetAllIconNames()
+        {
+            // add icons from plugins
+            var appPlugins = new DirectoryInfo(IOHelper.MapPath(SystemDirectories.AppPlugins));
+            var pluginIcons = appPlugins.Exists == false
+                ? new List<FileInfo>()
+                : appPlugins.GetDirectories()
+                    // Find all directories in App_Plugins that are named "Icons" and get a list of SVGs from them
```

Comment:
This could have been made MUCH faster if a real convention was created, like a specific Icon folder within an app plugin like App_Plugin/MyPlugin/BackOffice/Icons but now we have to deep scan every single nested folder in all app plugins which is very costly. I realize we are now having to add caching to this due to performance but this would be one of the reasons why. Being able to just look at the top level app_plugins folders means you'd only iterate as many top level foldres that exist which would be very quick.

This can also be made faster by not allocating DirectoryInfo objects for every folder that is looked at. Use Directory.EnumerateDirectories instead.

It would be ideal to have a real convention for this - at least documented now so that we can make this a real convention with better performance in netcore.

//cc @nul800sebastiaan + @kjac

Reply:
Yep, I realized that this was not the convention (thought it would be `App_Plugin/MyPlugin/Icons` but `App_Plugin/MyPlugin/BackOffice/Icons` is better! 👍 Let's go for that.

Reply:
@Shazwazza agreed. When I was trying to work out the performance issues with icons, I briefly changed this code to simply get all SVG files under app_plugins, and then do a reverse lookup for each file path to see if they were in an "Icons" folder. It was a slight improvement in comparison (since it didn't have to do multiple enumerations over the file system), but still not really good.

This however is already out in the wild. Making a convention at this point would be a breaking change. Question is if we should simply live with this in 8 and make the breaking change in netcore? Or rip the bandaid off now?

Reply:
I though it would only find icon-folders in the root plugin folder.

I say we should rip the bandaid and get the convention in. I noticed @nul800sebastiaan removed the 8.8.1 tag, so maybe it was postponed to 8.8.2 anyway?

Reply:
Yes, has been postponed to 8.8.2! No bandaid needs to be ripped, this code is unreleased at the moment, so all good to get it done peacefully! :-)

Reply:
I wonder if it should be possible to have a second level of subfolders inside the "Icons" folder or if it would cause performance issues?

For example if using these icons https://iconhub.io they are grouped in subfolders.

![image](https://user-images.githubusercontent.com/2919859/98438418-4fe86b00-20ea-11eb-9a33-645bf4ed28d7.png)

We can of course select the specific ones needed unless you want to create a general package to re-use in projects.

It could be nice e.g. to map a specific folder with a name/category (maybe via a package.manifest) so e.g. in icon picker it could should a dropdown to filter icons in this category - a bit similar the property editors, which belong to a group 😎

Just an idea for future improvements 😁🦄

Reply:
@bjarnef we've already learned from this PR that scanning for icons in folders is expensive and a serious performance concern, it's been rolled back and not released in the current releases yet. 

We're talking about icon metadata here: https://github.com/umbraco/Umbraco-CMS/issues/9122 - that should account for grouping in categories (and maybe even multilingual categories) as well.